src/edges.c

```diff
@@ -90,42 +112,62 @@ validate_single_region_edge(int *valid_edge,
 	 * the region borders for aligned edges only.
 	 */
 
-	bool lesser = direction == VIEW_EDGE_LEFT || direction == VIEW_EDGE_UP;
+	enum wlr_edges opposing = WLR_EDGE_NONE;
+
+	switch (direction) {
+	case WLR_EDGE_TOP:
+		opposing = WLR_EDGE_BOTTOM;
+		break;
+	case WLR_EDGE_BOTTOM:
+		opposing = WLR_EDGE_TOP;
+		break;
+	case WLR_EDGE_LEFT:
+		opposing = WLR_EDGE_RIGHT;
+		break;
+	case WLR_EDGE_RIGHT:
+		opposing = WLR_EDGE_LEFT;
+		break;
+	default:
+		/* Should never be reached */
+		assert(false);
```

Comment:
I think we had issues with this before when built without asserts (and should likely add an optimized build to the CI).

AFAIR the conclusion was to either
- handle all possible cases and thus get a compile error when we missed a case (`../src/edges.c:45:2: error: enumeration value ‘WLR_EDGE_NONE’ not handled in switch [-Werror=switch]`)
- or keeping the `default` and `assert()` but also `break` or `return` (or even `abort()`) with or without additional logging depending on the case

I am not sure if we could catch this via checkpatch.pl for the future.

There seem to be another `default: assert(false)` in edges.c.

Reply:
Good point. In this case, a return is OK because it will effectively skip validation of an ill-defined edge. In `build_edge`, getting an edge wrong is a completely unacceptable case, so I've added an `abort()` with a descriptive message. We only ever call `build_edge` with hard-coded arguments (and only ever should), so hard failing in all cases seems acceptable.

Reply:
> and should likely add an optimized build to the CI to detect this

done.